packages/core/src/services/ai/index.ts

```diff
@@ -94,12 +94,23 @@ export async function ai({
     ...(aiSdkProvider || {}),
   }
   try {
-    const rule = applyCustomRules({
+    const rule = applyAllRules({
       providerType: apiProvider.provider,
       messages: originalMessages,
       config: originalConfig,
     })
 
+    if (rule.rules.length > 0) {
+      return Result.error(
+        new ChainError({
+          code: RunErrorCodes.AIRunError,
+          message:
+            'There are rule violations:\n' +
+            rule.rules.map((rule) => `- ${rule.ruleMessage}`).join('\n'),
+        }),
+      )
+    }
```

Comment:
Why is this necessary now? How were the rules returned before when some rule failed here?

Reply:
We never checked for rules before, so it crashed